### PR TITLE
Efficiency

### LINKED ISSUE #13: Reduce overhead automatically
The code below shows that for the simplest of generators, there's quite an overhead. This is to be expected as there's not much content in the loop itself.

The 6x increase isn't that bad (especially when compared to the de-facto `ProgressBar`, hah). However, this is reduced to about 3x if we set `miniters` to be high enough that it limits update more than the `mininterval`. This is due to the internal time updating of the function.

I think it's best to attempt to set a higher `miniters` by default (maybe based on iterator arguments, like `max(int(args[0] / 100), 1)`) or at least document this effect somewhere...

Still. 100x faster that `ProgressBar` in some scenarios. And prettier by default. Nice :+1: 

`xrange`: 3.02 sec
`tqdm.trange`: 19.6 sec
`tqdm.trange, miniters=niters/100`: 9.71 sec
`progressbar.ProgressBar()(xrange)`: 87.8 sec

```{python}
#!/usr/bin/env python2.7

# Profiling/overhead tests
from timeit import timeit


stmts = ('[i for i in xrange(int(1e8))]',
    'import tqdm; [i for i in tqdm.trange(int(1e8))]',
    'import tqdm; [i for i in tqdm.trange(int(1e8), miniters=int(1e6))]',
    'import progressbar; [i for i in progressbar.ProgressBar()(xrange(int(1e8)))]')
for s in stmts:
    print s
    print timeit(stmt=s, number=1), 'seconds'
```

### PR DESCRIPTION
**Pull request by casperdcl**
fixes #13

### CODE REVIEW

**Comment by hadim** [comment]
Very nice !

One minor comment : I prefer not to put 2.0 version in the setup.py for now. Let see what happend whith the project and pypi admin and then we could choose to release a new version.

I am also in favor to release a 1.1 version since there is no breaking API changes (and it's better like that).

**Comment by hadim** [comment]
Last thing : travis report 96% coverage. Could you also add tests to make it 100% ?

See : https://travis-ci.org/tqdm/tqdm/jobs/67383090#L131

**Comment by casperdcl** [comment]
Cool.

Do you want to merge and make the 1.1 change, or should I?

**Comment by casperdcl** [comment]
Thought it was 100 when I checked. Oops.

**Comment by hadim** [comment]
The branch is on your repo so I let you do it.

By the way next time you can add branch in this repo directly since I added you as owner so everyone can easly push in a branch and update PR.

**Comment by hadim** [comment]
I also propose that the author of a PR never merge his own PR and always let another contributor do it.

**Comment by casperdcl** [comment]
yeah but it doesn't matter in rare cases (e.g. one minor comment for
improvement) where the original author can make the change and merge.
Anyway I have to add a test so yes obvs you merge.

Btw do you want me to look at the other PRs?

**Comment by hadim** [comment]
I think #8 is almost ready to merge, feel free to do it (maybe a minor modification is need). As for #9, it needs more works, I will do it next week.

**Comment by hadim** [comment]
LGTM for me

File: setup.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 setup(
     name='tqdm',
-    version='1.0',
+    version='1.1',
```
Comment (kmike) [inline setup.py]: I think it is better to change version number in a separate commit, with a git tag and release notes.

**Pull request merged**